frappe/core/doctype/communication/communication.py

```diff
@@ -173,9 +173,9 @@ def set_signature_in_email_content(self):
 		html_signature = soup.find("div", {"class": "ql-editor read-mode"})
 		_signature = None
 		if html_signature:
-			_signature = html_signature.renderContents()
+			_signature = html_signature.renderContents(encoding='utf-8')
 
-		if (_signature or signature) not in self.content:
+		if (_signature.decode("utf-8") or signature) not in self.content:
```

Comment:
Not sure how is this any better. 


line 174 makes it `None` so you'll be attempting decode on `None`? 

---

Use `cstr` from `frappe.utils` to handle all the cases.

Reply:
cstr converts most things to python string automatically.